### ISSUE TITLE
{Linux] DotA 2 Reborn libs Error

### ISSUE THREAD

**Issue by 420git** [ValveSoftware/Dota-2#387]
So I have been playing DotA 2 and TF2 and other games on steam for the past 2 years using the native steam runtime after deleting some buggy libs.(Im on Arch with newer libs)

But DotA 2 Reborn didnt start at all, and I get the following error

libGL error: unable to load driver: r600_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: r600
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast

So I tried using my system libs with $ STEAM_RUNTIME=0 steam , and after installing most of the needed libs which steam asked for (I finally had to add alucryd repo because one lib couldnt be found) I ran steam and DotA2 TF2 work flawlessly but DotA 2 Reborn gives me the following error

failed to dlopen "librendersystemgl.so" error=libudev.so.0: wrong ELF class: ELFCLASS32
failed to dlopen "libpangoft2-1.0.so" error=libpng12.so.0: wrong ELF class: ELFCLASS32
Plat_GetProcAddresses: unable to load 'libpangoft2-1.0.so'

Now that DotA 2 has completly moved to reborn I cant play it at all.

Im on 64 with multilib enabled. Using Mesa drivers for AMD VGA

I tried using
$ STEAM_RUNTIME=0 LD_PRELOAD='/usr/lib32/libudev.so.0 /usr/lib32/libpng12.so.0' steam

which makes steam fail to load completely and gives me this error

ERROR: ld.so: object '/usr/lib32/libudev.so.0' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/usr/lib32/libpng12.so.0' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.

and

$ STEAM_RUNTIME=0 LD_PRELOAD='/usr/lib64/libudev.so.0 /usr/lib64/libpng12.so.0' steam

which fails to load them as well but steam is able to start

ERROR: ld.so: object '/usr/lib64/libudev.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib64/libpng12.so.0' from LD_PRELOAD cannot be preloaded (cannot open shared object file): ignored.

So I have no idea what I should do now.
Any suggestions ?

**Comment by fwsGonzo** [comment]
looks like the steam runtime is 64bit now, and its ignoring your 32bit ELF shared objects... so, I guess you will simply have to reinstall the whole thing and try to make nice with the 64bit versions

**Comment by danginsburg** [comment]
Can you try the workaround suggested in #35 of doing a symbolic link for the missing libs, i.e. ln -s /lib64/libudev.so.1 ~/.local/share/Steam/SteamApps/common/dota 2 beta/game/bin/linuxsteamrt64/libudev.so.0 and see if that works?

**Comment by 420git** [comment]
I tried the workaround in #35 and it didnt work.
I however fixed this thanks to alucryd.
Looks like DotA 2 Reborn is only 64bit now so it doesnt need the previous 32bit libs at all.
He just added libpng12 and libudev0 to his x86-64 repo which were missing from my lib64 folder and now it works flawlessly after I disable steamruntime.

I think I have another issue with steam runtime in which it cant load the r600 and swart drivers although I checked I have those installed in both lib32 and lib64.
I read somewhere that I could specify those drivers for opengl to load as symlinks but I didnt want to mess with my system so I didnt try it.
